feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -797,25 +772,83 @@ class BillingWrapper(
     }
 
     private fun buildPurchaseParams(
-        storeProduct: GoogleStoreProduct,
-        purchaseOption: GooglePurchaseOption,
+        storeProduct: StoreProduct,
+        purchaseOption: PurchaseOption?,
         replaceSkuInfo: ReplaceSkuInfo?,
-        appUserID: String,
-        callback: ()
-    ): BillingFlowParams {
+        appUserID: String
+    ): Result<BillingFlowParams, PurchasesError> {
+        val googleProduct = storeProduct.googleProduct
+
+        if (googleProduct == null) {
+            val error = PurchasesError(
+                PurchasesErrorCode.UnknownError,
+                PurchaseStrings.INVALID_STORE_PRODUCT_TYPE.format(
+                    "Play",
+                    "GoogleStoreProduct"
+                )
+            )
+            errorLog(error)
+            return Result.Error(error)
+        }
+
+        return if (googleProduct.type == ProductType.SUBS) {
+            buildSubscriptionPurchaseParams(googleProduct, purchaseOption, replaceSkuInfo, appUserID)
+        } else {
+            buildOneTimePurchaseParams(googleProduct, appUserID)
```

Comment:
I separated them since subscriptions have a few extra pieces of logic. I could add checks around but it felt better to just separate the logic.